### PR TITLE
refractor PageConvertor, support int, long, double type

### PR DESCRIPTION
**Pull request by yyang52**
Test plan - (Please fill in how you tested your changes)

Please make sure your submission complies with our [Development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [Formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), and [Commit Message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests) guidelines. Don't forget to follow our [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution) for any code copied from other projects.

Fill in the release notes towards the bottom of the PR description.
See [Release Notes Guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) for details.

```
== RELEASE NOTES ==

General Changes
* ...
* ...

Hive Changes
* ...
* ...
```

If release note is NOT required, use:

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by yyang52** [comment]
@jikunshang @winningsix 
Please take a review thanks.

File: presto-common/src/main/java/com/facebook/presto/common/PageConvertor.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+                }
+            }
+            else if (types[i].equals(DOUBLE_TYPE_)) {
+                sizeForEachBlock[i] = 8;
+                for (int j = 0; j < positionCount; j++) {
+                    _UNSAFE.putDouble(dataBuffers[i] + j * sizeForEachBlock[i], longBitsToDouble(block.getLong(j)));
+                }
+            }
+            else {
+                throw new UnsupportedOperationException("block type " + block.getClass().getName() + " not supported!!");
```
Comment (jikunshang) [inline presto-common/src/main/java/com/facebook/presto/common/PageConvertor.java:81]: `block.getClass().getName()` -> `types[i]` ?

File: presto-common/src/test/java/com/facebook/presto/common/TestPageConvertor.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        assertEquals(ret.length, 3);
+        long dataBuffer0 = ret[0];
+        for (int i = 0; i < entries; i++) {
+            assertEquals(PageConvertor.getUnsafe().getLong(dataBuffer0 + i * Long.BYTES), (long) i);
+        }
+        PageConvertor.destroyOffHeapBuffer(ret);
+    }
+
+    @Test
+    public void testdOUBLE()
```
Comment (jikunshang) [inline presto-common/src/test/java/com/facebook/presto/common/TestPageConvertor.java:49]: typo

File: presto-common/src/test/java/com/facebook/presto/common/TestPageConvertor.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    {
+        int entries = 10;
+        String[] types = {"Double", "Double", "Double"};
+        BlockBuilder blockBuilder = DOUBLE.createBlockBuilder(null, entries);
+        for (int i = 0; i < entries; i++) {
+            DOUBLE.writeDouble(blockBuilder, i);
```
Comment (jikunshang) [inline presto-common/src/test/java/com/facebook/presto/common/TestPageConvertor.java:55]: `(double)i`

File: presto-common/src/main/java/com/facebook/presto/common/PageConvertor.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+import com.facebook.presto.common.block.Block;
+import com.facebook.presto.common.block.BlockBuilder;
+import sun.misc.Unsafe;
+
+import java.lang.reflect.Field;
+
+import static com.facebook.presto.common.type.BigintType.BIGINT;
+import static com.facebook.presto.common.type.DoubleType.DOUBLE;
+import static com.facebook.presto.common.type.IntegerType.INTEGER;
+import static java.lang.Double.longBitsToDouble;
```
Comment (jikunshang) [inline presto-common/src/main/java/com/facebook/presto/common/PageConvertor.java:25]: I think we can remove this import, use `Double.longBitsToDouble()` instead, which is more readable for other guys(I feel a little confused when I review Line 77). Same for `Double.doubleToLongBits()`.

File: presto-common/src/main/java/com/facebook/presto/common/PageConvertor.java [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+            long bufferSize = block.getSizeInBytes();
+            dataBuffers[i] = _UNSAFE.allocateMemory(bufferSize);
+            if (types[i].equals(INT_TYPE_)) {
+                sizeForEachBlock[i] = 4;
+                for (int j = 0; j < positionCount; j++) {
+                    _UNSAFE.putInt(dataBuffers[i] + j * sizeForEachBlock[i], block.getInt(j));
+                }
+            }
+            else if (types[i].equals(LONG_TYPE_)) {
+                sizeForEachBlock[i] = 8;
```
Comment (jikunshang) [inline presto-common/src/main/java/com/facebook/presto/common/PageConvertor.java:69]: necessary to check block type here? eg: what if passed in an `IntArrayBlock`

**Review by jikunshang — COMMENTED** [review]
Overall LGTM.

File: presto-common/src/main/java/com/facebook/presto/common/PageConvertor.java (hunk 1 above)
Comment (yyang52) [reply presto-common/src/main/java/com/facebook/presto/common/PageConvertor.java]: fixed

File: presto-common/src/test/java/com/facebook/presto/common/TestPageConvertor.java (hunk 2 above)
Comment (yyang52) [reply presto-common/src/test/java/com/facebook/presto/common/TestPageConvertor.java]: fixed

File: presto-common/src/test/java/com/facebook/presto/common/TestPageConvertor.java (hunk 3 above)
Comment (yyang52) [reply presto-common/src/test/java/com/facebook/presto/common/TestPageConvertor.java:55]: (IDEA suggested) casting here is redundant

File: presto-common/src/main/java/com/facebook/presto/common/PageConvertor.java (hunk 4 above)
Comment (yyang52) [reply presto-common/src/main/java/com/facebook/presto/common/PageConvertor.java:27]: Presto used import (in DoubleType)  should we follow that or code as you suggested?

File: presto-common/src/main/java/com/facebook/presto/common/PageConvertor.java (hunk 5 above)
Comment (yyang52) [reply presto-common/src/main/java/com/facebook/presto/common/PageConvertor.java:71]: added

File: presto-common/src/main/java/com/facebook/presto/common/PageConvertor.java (hunk 4 above)
Comment (jikunshang) [reply presto-common/src/main/java/com/facebook/presto/common/PageConvertor.java:27]: OK. Let's keep it.

**Review by jikunshang — APPROVED** [review]

**Pull request merged**